tracer/src/Datadog.Tracer.Native/fault_tolerant_rewriter.cpp

```diff
@@ -0,0 +1,594 @@
+        instrumentationVersion.c_str(), static_cast<ULONG>(instrumentationVersion.length()),
+        &instrumentationVersionIdToken);
+
+    if (FAILED(hr))
+    {
+        Logger::Warn("*** FaultTolerantRewriter::InjectSuccessfulInstrumentation() Failed to define instrumentation "
+                     "version string ",
+                     moduleId, " ", methodId);
+        return hr;
+    }
+
+    // static void AddSuccessfulInstrumentationVersion(IntPtr moduleId, int methodToken, string instrumentationVersion,
+    // InstrumentingProducts products)
+
+    if (sizeof(UINT_PTR) == 4) // 32-bit
+    {
+        rewriterWrapper.LoadInt32(static_cast<INT32>(moduleId));
+    }
+    else if (sizeof(UINT_PTR) == 8) // 64-bit
+    {
+        rewriterWrapper.LoadInt64(static_cast<INT64>(moduleId));
+    }
+
+    rewriterWrapper.LoadInt32(methodId);
+    rewriterWrapper.LoadStr(instrumentationVersionIdToken);
+    rewriterWrapper.LoadInt32(static_cast<INT32>(instrumentingProduct));
+    ILInstr* reportSuccessfulInstrumentation;
+    faultTolerantTokens->WriteReportSuccessfulInstrumentation(&rewriterWrapper, &reportSuccessfulInstrumentation);
+
+    hr = rewriter.Export();
+
+    if (IsDumpILRewriteEnabled())
+    {
+        std::string original_code = m_corProfiler->GetILCodes(
+            "*** FaultTolerantRewriter::InjectSuccessfulInstrumentation() Original Code: ", &rewriter, *caller,
+            moduleHandler->GetModuleMetadata()->metadata_import);
+        Logger::Info(original_code);
+    }
+
+    if (FAILED(hr))
+    {
+        Logger::Warn("Failed to emit IL for injecting successful instrumentation, ModuleID=", moduleId);
+        return hr;
+    }
+
+    Logger::Info("Successfully added injection of successful instrumentation report, moduleId = ", moduleId);
+    return hr;
+}
+
+HRESULT FaultTolerantRewriter::RewriteInternal(RejitHandlerModule* moduleHandler,
+                                               RejitHandlerModuleMethod* methodHandler,
+                                               ICorProfilerFunctionControl* pFunctionControl)
+{
+    if (!is_fault_tolerant_instrumentation_enabled)
+    {
+        return m_methodRewriter->Rewrite(moduleHandler, methodHandler, pFunctionControl);
+    }
+
+    const auto moduleId = moduleHandler->GetModuleId();
+    const auto methodId = methodHandler->GetMethodDef();
+
+    if (FaultTolerantTracker::Instance()->IsKickoffMethod(moduleId, methodId))
+    {
+        const auto instrumentationVersion = m_methodRewriter->GetInstrumentationVersion(moduleHandler, methodHandler);
+        const auto instrumentingProduct = m_methodRewriter->GetInstrumentingProduct(moduleHandler, methodHandler);
+
+        if (FaultTolerantTracker::Instance()->IsInstrumentationVersionSucceeded(moduleId, methodId, instrumentationVersion, instrumentingProduct))
+        {
+            // Request Revert for the original method
+            //std::vector<MethodIdentifier> requests = {{moduleId, methodId}};
```

Comment:
Do you still need these comments?

Reply:
Nope, I have all kind of left overs that I'll clean up. They're all from experiments and games I did.